app/controllers/saml_idp_controller.rb

```diff
@@ -7,14 +7,16 @@ class SamlIdpController < ApplicationController
   include SamlIdpAuthConcern
   include SamlIdpLogoutConcern
   include FullyAuthenticatable
+  include VerifyProfileConcern
 
   skip_before_action :verify_authenticity_token
   skip_before_action :handle_two_factor_authentication, only: :logout
 
   def auth
     return confirm_two_factor_authenticated(@request_id) unless user_fully_authenticated?
-    process_fully_authenticated_user do |needs_idv|
-      return store_location_and_redirect_to_verify_url if needs_idv
+    process_fully_authenticated_user do |needs_idv, needs_profile_finish|
+      return store_location_and_redirect_to(verify_url) if needs_idv
+      return store_location_and_redirect_to(profile_or_verify_profile_url) if needs_profile_finish
```

Comment:
If I'm reading the logic correctly, if `needs_profile_finish` is true, then `profile_or_verify_profile_url` will always be `verify_profile_route`, so might as well just go there directly without going through the logic, right?